### ISSUE TITLE
[Feature Request] - Add Depth value search and expand Weirdness search limits

### ISSUE THREAD

**Issue by Jereaux** [Cubitect/cubiomes-viewer#181]
The depth biome layer/scale limit map is very useful for finding certain features of seeds with deep lakes or mountain formations. Adding a search feature in climate parameters to search against depth could be useful for some specific types of searches.  While I do not know the true highest or lowest depth value, I would appreciate being able to search down to a depth of -9000 before it switches to -Inf if this could be implemented. The lowest I have found thus far is down to -8700 but I'm not sure how much further it could go as that was 1 block above the void. 

I would also like to request a change to the upper and lower limit for searching Weirdness. Currently the UI allows for searches to +-9334 weirdness before switching to +-Inf, however more extreme weirdness produces unique features. Expanding the limits to allow for searches down to -22000 and up to 22000 would be useful. This type of search already works as I was able to edit session files to accomplish this, however it does require quite a bit more work than utilizing a slider.

And just because I have never been able to say it directly, thank you so much for all the work you have done on this project. I thoroughly enjoy using cubiomes-viewer, and have made many fun discoveries through this tool that I have been able to share. I appreciate how much you have individually contributed to the Minecraft community through your work. Thank you!

**Comment by Grills1** [comment]
I agree with Jereaux, and stumbled upon this request while researching the same thing. There are several instances where the an incredibly low depth (<-5000) and high weirdness (15,000+) leads to unique features in the overworld, in addition to allowing otherwise impossible structure generations in Bedrock.   I am sure that either I or Jereaux could provide plenty of seed and coordinate examples upon request.

Thank you so much for everything you do!

**Comment by KavyanshKhaitan2** [comment]
just watched TheMisterEpic's video and agree with Jereaux

**Comment by Delvin4519** [comment]
Support this feature.

Increase range of sliders. I'd say up to +32767 and -32768.

**Comment by Delvin4519** [comment]
Limits are still not expanded as of relase v3.0.0. So it still required editing cubiomes session.save files to obtain these in-game possible values.

**Comment by Cubitect** [comment]
The limits for the climate parameter sliders are not arbitrary, but reflect the minimum and maximum values that can affect the biome generation. From what I can tell, it will be more useful to have a new dedicated filter that can locate a minimum or maximum for a single climate value in an area, which should be faster and easier to use than fiddling with the sliders. I have added such a filter in v3.0 ("Locate climate extreme").

The depth value is not a proper climate parameter, but an emergent value from the continentalness, erosion and weirdness and the gradient descent algorithm, that is used for the other climate parameter searches, is not suitable for it. So a depth search would need a completely new implementation and will be much slower than the other climate parameters. Instead I added a surface height approximation filter in v3.0, which allows you to check the resulting height, also for older MC versions, but it only checks a specified location, not an area.
